### PR TITLE
ReplicateBlob replicates tombstone.

### PR DESCRIPTION
**Pull request by JingQianCloud**

### CODE REVIEW

File: ambry-protocol/src/main/java/com/github/ambry/protocol/AmbryRequests.java [hunk 1]
```diff
@@ -863,6 +883,76 @@ public void handleReplicateBlobRequest(NetworkRequest request) throws IOExceptio
...
+      AdminResponseWithContent adminResponse = AdminResponseWithContent.readFrom(stream);
+      if (adminResponse.getError() != ServerErrorCode.No_Error) {
+        logger.error("ReplicateBlobRequest failed to get tombstone {} from the remote node {} {} {}", blobId,
+            remoteHostName, remoteHostPort, adminResponse.getError());
+        return new Pair(adminResponse.getError(), null);
+      }
+
+      byte[] jsonBytes = adminResponse.getContent();
+      ObjectMapper objectMapper = new ObjectMapper();
+      StoreKeyJacksonConfig.setupObjectMapper(objectMapper, new BlobIdFactory(clusterMap));
```
Comment (justinlin-linkedin) [inline ambry-protocol/src/main/java/com/github/ambry/protocol/AmbryRequests.java:927]: there two variables can be class members and only need to be initialized once.

File: ambry-protocol/src/main/java/com/github/ambry/protocol/AmbryRequests.java [hunk 2]
```diff
@@ -863,6 +883,76 @@ public void handleReplicateBlobRequest(NetworkRequest request) throws IOExceptio
...
+      Map<String, MessageInfo> messages =
+          objectMapper.readValue(jsonBytes, new TypeReference<Map<String, MessageInfo>>() {
+          });
+      if (messages == null || messages.size() != 1) {
+        logger.error("ReplicateBlobRequest adminRequest for {} from the remote node {} {} return {} entries", blobId,
```
Comment (justinlin-linkedin) [inline ambry-protocol/src/main/java/com/github/ambry/protocol/AmbryRequests.java:932]: can you also print out the content of the map so we know what went wrong.

File: ambry-server/src/integration-test/java/com/github/ambry/server/ServerHttp2Test.java [hunk 3]
```diff
@@ -131,6 +131,13 @@ public void replicateBlobCaseTest() {
         notificationSystem, writeRepair);
   }
 
+  @Test
+  public void replicateDeleteRecrodTest() {
```
Comment (justinlin-linkedin) [inline ambry-server/src/integration-test/java/com/github/ambry/server/ServerHttp2Test.java:135]: typo on method name

File: ambry-protocol/src/main/java/com/github/ambry/protocol/AmbryRequests.java [hunk 4]
```diff
@@ -828,7 +832,23 @@ public void handleReplicateBlobRequest(NetworkRequest request) throws IOExceptio
                 remoteHostPort);
             errorCode = ServerErrorCode.No_Error;
           } // if (output == null)
-        } // if (errorCode == ServerErrorCode.No_Error)
+        } else if (errorCode == ServerErrorCode.Blob_Deleted) {
```
Comment (justinlin-linkedin) [inline ambry-protocol/src/main/java/com/github/ambry/protocol/AmbryRequests.java:835]: can we create a metric for this one?

File: ambry-protocol/src/main/java/com/github/ambry/protocol/AmbryRequests.java (hunk 1 above)
Comment (JingQianCloud) [reply ambry-protocol/src/main/java/com/github/ambry/protocol/AmbryRequests.java]: Changed.

File: ambry-protocol/src/main/java/com/github/ambry/protocol/AmbryRequests.java (hunk 2 above)
Comment (JingQianCloud) [reply ambry-protocol/src/main/java/com/github/ambry/protocol/AmbryRequests.java]: Added the dump.

File: ambry-server/src/integration-test/java/com/github/ambry/server/ServerHttp2Test.java (hunk 3 above)
Comment (JingQianCloud) [reply ambry-server/src/integration-test/java/com/github/ambry/server/ServerHttp2Test.java]: fixed the typo.

File: ambry-protocol/src/main/java/com/github/ambry/protocol/AmbryRequests.java (hunk 4 above)
Comment (JingQianCloud) [reply ambry-protocol/src/main/java/com/github/ambry/protocol/AmbryRequests.java:838]: Added a meter for it.

**Review by justinlin-linkedin — APPROVED** [review]

**Review by SophieGuo410 — APPROVED** [review]

**Review by justinlin-linkedin — APPROVED** [review]

**Pull request merged**